cpp/inc/bond/core/blob.h

```diff
@@ -87,42 +81,36 @@ class blob
     template <typename T, template <typename U> class SmartPtr>
     blob(const SmartPtr<T>& buffer, uint32_t length)
         : _buffer(wrap_in_shared_ptr(buffer)),
-          _content(_buffer.get()),
           _length(length)
     {
-        bond::detail::checked_add(_content, length);
+        bond::detail::checked_add(_buffer.get(), length);
     }
 
     /// @brief Construct from a smart pointer other than boost::shared_ptr
     ///
     /// Not recommended for performance reasons. Use boost::shared_ptr whenever possible.
     template <typename T, template <typename U> class SmartPtr>
     blob(const SmartPtr<T>& buffer, uint32_t offset, uint32_t length)
-        : _buffer(wrap_in_shared_ptr(buffer)),
-          _content(bond::detail::checked_add(_buffer.get(), offset)),
+        : _buffer(wrap_in_shared_ptr(buffer, offset)),
           _length(length)
     {
-        bond::detail::checked_add(_content, length);
+        bond::detail::checked_add(_buffer.get(), length);
```

Comment:
Use `content()` or `data()` instead of `_buffer.get()` in most places?

Reply:
Outside of the scope of this PR. The whole class needs a clean and modern rewrite.